Fixes #98: Don't try to print non-printable chars

abishekvashok opened the pull request:
This lead to unintended consequences as random block of characters
appeared on resizing when we tried to pass non printable -1 to addch.
Introduces a simple if statement to fix it.

Here's a gif:

![](https://im6.ezgif.com/tmp/ezgif-6-cdb4275d1d6d.gif)

abishekvashok committed:
Fixes #98: Properly initialise head to false

This fix introduces a loop to make sure that the head is properly
initialised so that random block characters don't appear on screen when
resizing.

Fixes #98
Signed-off-by: Abishek V Ashok <abishekvashok@gmail.com>

StavromulaBeta commented:
I'm still seeing a single line of block characters on the bottom row of my terminal emulator when I quickly shrink the window. However, it is certainly much better than before, where the blocks covered the entire terminal window in vertical lines. I'm using Xterm on Debian.

abishekvashok committed:
Fixes #98: Don't try to print non-printable chars

This lead to unintended consequences as random block of characters
appeared on resizing when we tried to pass non printable -1 to addch.
Introduces a simple if statement to fix it.

Fixes #98
Signed-off-by: Abishek V Ashok <abishekvashok@gmail.com>

abishekvashok commented:
> I'm still seeing a single line of block characters on the bottom row of my terminal emulator when I quickly shrink the window. However, it is certainly much better than before, where the blocks covered the entire terminal window in vertical lines. I'm using Xterm on Debian.

Better isn't something we want, so I peered in deeper and found out that non-printable characters were being fed to ncurses, fixed it this time :) [the head on the other hand seemed to suppress it a bit]

I've updated the PR now @StavromulaBeta, can you make sure it works on your end?

abishekvashok commented:
I don't like to self merge without reviews, but seems like it would take some time and it works on some machines I had access to. So doing it :D
